yjit/src/backend/x86_64/mod.rs

```diff
@@ -164,6 +164,35 @@ impl Assembler
 
                     asm.push_insn(op, vec![opnd0, opnd1], target, text, pos_marker);
                 },
+                Op::Shl | Op::Shr => {
+                    let (opnd0, opnd1) = match (opnds[0], opnds[1]) {
+                        // Instruction output whose live range spans beyond this instruction
+                        (Opnd::InsnOut { .. }, _) => {
+                            let idx = match original_opnds[0] {
+                                Opnd::InsnOut { idx, .. } => {
+                                    idx
+                                },
+                                _ => panic!("nooooo")
```

Comment:
```suggestion
                                _ => unreachable!()
```

Smaller binary and just as descriptive :p